### PR TITLE
Catch inf and nan in divisions

### PR DESCRIPTION
**Pull request by lochel**
see [ticket:4230](https://trac.openmodelica.org/OpenModelica/ticket/4230)

### CODE REVIEW

**Comment by OpenModelica-Hudson** [comment]
The job failed to compile; for details, see [OpenModelica_TEST_PULL_REQUEST 2017-01-25_16-11-09](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3780/console).

**Comment by OpenModelica-Hudson** [comment]
The job failed to compile; for details, see [OpenModelica_TEST_PULL_REQUEST 2017-01-25_16-17-22](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3781/console).

**Comment by OpenModelica-Hudson** [comment]
The job failed to compile; for details, see [OpenModelica_TEST_PULL_REQUEST 2017-01-25_16-28-47](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3782/console).

**Comment by OpenModelica-Hudson** [comment]
The job failed to compile; for details, see [OpenModelica_TEST_PULL_REQUEST 2017-01-25_16-32-08](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3783/console).

**Comment by OpenModelica-Hudson** [comment]
The job failed to compile; for details, see [OpenModelica_TEST_PULL_REQUEST 2017-01-25_16-39-35](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3784/console).

**Comment by OpenModelica-Hudson** [comment]
The job failed to compile; for details, see [OpenModelica_TEST_PULL_REQUEST 2017-01-25_16-46-15](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3785/console).

**Comment by OpenModelica-Hudson** [comment]
The job failed to compile; for details, see [OpenModelica_TEST_PULL_REQUEST 2017-01-25_17-38-48](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3786/console).

**Comment by OpenModelica-Hudson** [comment]
The job failed to compile; for details, see [OpenModelica_TEST_PULL_REQUEST 2017-01-25_19-00-44](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3787/console).

**Comment by OpenModelica-Hudson** [comment]
The job failed to compile; for details, see [OpenModelica_TEST_PULL_REQUEST 2017-01-26_10-07-54](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3795/console).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2017-01-26_10-50-30](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3796/).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2017-01-26_11-08-32](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3797/).

**Comment by lochel** [comment]
It seems that there a some models in the test suite that calculate with `inf` or `nan`. I don't think that this is really expected, right?

File: SimulationRuntime/c/util/division.h [hunk 1]
```diff
@@ -41,23 +42,29 @@
...
+{
+  modelica_real res;
+  if(b != 0.0)
+    res = a/b;
+  else if(initial_ && a == 0.0)
+    res = 0.0;
+  else
+    res = a / division_error_equation_time(threadData, a, b, msg, equationIndexes, time_, noThrowDivZero);
+
+  if(!valid_number(res))
```
Comment (vruge) [inline SimulationRuntime/c/util/division.h]: if(noThrowDivZero && !valid_number(res))

**Comment by OpenModelica-Hudson** [comment]
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2017-01-26_13-42-14](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3802/console).

**Comment by vruge** [comment]
it's seem the check about if(b != 0.0) ... not really needed?!
please rember https://trac.openmodelica.org/OpenModelica/ticket/3568.

**Comment by lochel** [comment]
We should just remove all of the division macros.

**Pull request closed without merging**